### PR TITLE
Use frozen-partial semantics during blocking operations.  That way we won't block users as much.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Tagging @dotnet/roslyn-ide for thoughts on this.

### CODE REVIEW

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Backspace.cs [hunk 1]
```diff
@@ -75,8 +75,7 @@ private void ExecuteBackspaceOrDelete(ITextView textView, Action nextHandler, bo
             else
             {
                 var textBeforeDeletion = SubjectBuffer.AsTextContainer().CurrentText;
-                var documentBeforeDeletion = textBeforeDeletion.GetDocumentWithFrozenPartialSemanticsAsync(CancellationToken.None)
-                                                               .WaitAndGetResult(CancellationToken.None);
+                var documentBeforeDeletion = textBeforeDeletion.GetDocumentWithFrozenPartialSemantics(CancellationToken.None);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Backspace.cs]: this was bad before.  completion was blocking on async code on backspace.

File: src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`2.VsLanguageDebugInfo.cs [hunk 2]
```diff
@@ -278,7 +278,7 @@ public int ValidateBreakpointLocation(IVsTextBuffer pBuffer, int iLine, int iCol
                 if (textBuffer != null)
                 {
                     var snapshot = textBuffer.CurrentSnapshot;
-                    Document document = snapshot.AsText().GetDocumentWithFrozenPartialSemanticsAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+                    Document document = snapshot.AsText().GetDocumentWithFrozenPartialSemantics(cancellationToken);
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`2.VsLanguageDebugInfo.cs]: Blocking on async on a UI thread operation.  This is likely why people might have seen breakpoint resolution modal dialogs lasting a long time.

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.Session_ComputeModel.cs [hunk 3]
```diff
@@ -96,7 +96,7 @@ public void Do()
                         }
 
                         // get partial solution from background thread.
-                        _documentOpt = await _text.GetDocumentWithFrozenPartialSemanticsAsync(cancellationToken).ConfigureAwait(false);
```
Comment (rchande) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.Session_ComputeModel.cs]: Why?

Comment (rchande) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.Session_ComputeModel.cs]: Oh, I see.

**Comment by CyrusNajmabadi** [comment]
Tagging @Pilchie @jasonmalinowski

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide @Pilchie

**Comment by CyrusNajmabadi** [comment]
> The platform team changed copy to not use accurate tagging (reserving it for "Print", etc), so I think we should not take this change.

Ok.  But i'm still going to amke the change for getting frozen-partial to be synchronous.  The features that need frozen-partial are synchronous already, and they should be parsing the tree on their thread, instead of blocking on a threadpool task.

**Comment by CyrusNajmabadi** [comment]
@Pilchie Plz review again.

**Comment by CyrusNajmabadi** [comment]
retest windows_release_unit64_prtest  please

**Comment by CyrusNajmabadi** [comment]
retest windows_release_unit64_prtest  please

**Comment by jasonmalinowski** [comment]
Nice find!

**Pull request merged**